components/public-api/gitpod/experimental/v1/tokens.proto

```diff
@@ -0,0 +1,88 @@
+syntax = "proto3";
+
+package gitpod.experimental.v1;
+
+import "google/protobuf/timestamp.proto";
+
+service TokenService {
+
+    // CreateToken creates a new token and returns it.
+    rpc CreateToken(CreateTokenRequest) returns (CreateTokenResponse) {}
+
+    // GetToken returns one token.
+    rpc GetToken(GetTokenRequest) returns (GetTokenResponse) {}
+
+    // ListTokens returns a list of tokens.
+    rpc ListTokens(ListTokensRequest) returns (ListTokensResponse) {}
+
+    // UpdateToken either regenerates a token or updates a token's scopes.
+    rpc UpdateToken(UpdateTokenRequest) returns (UpdateTokenResponse) {}
+
+    // DeleteToken removes a token.
+    rpc DeleteToken(DeleteTokenRequest) returns (DeleteTokenResponse) {}
+}
+
+message CreateTokenRequest {
+    string token_hash = 1;
+    string user_id = 2;
+    string name = 3;
+    string description = 4;
+    string expiration_time = 5;
+    repeated Scope scopes = 6;
+    google.protobuf.Timestamp created_at = 7;
+}
```

Comment:
```suggestion
message CreateTokenRequest {
    Token token = 1;
}
```

You can do this. It's okay if not every field is used in the creation, you can comment on individual fields if they are read only or not.